File: java/bucles-for.java
import java.util.ArrayList;
import java.util.HashMap;
import java.util.List;
import java.util.Map;

public class Main {
    public static void main(String[] args) {
        
        String[] nombres = {
            "camila", "sebastian R", "Cristian", "Andres R", "Anyelo", 
            "Esteban A", "Felipe", "Fito", "Hanssy R", "Hanz l", 
            "Jhon", "juan pablo", "Nilo", "Santiago", "Santiago S", "Yuri \n\n"
        };

       
        for (String nombre : nombres) {
            System.out.println(nombre);
        }

        
        List<Map<String, Object>> personas = new ArrayList<>();

       
        Map<String, Object> a = new HashMap<>();
        a.put("nombre", "sebastian R");
        a.put("Edad", 24);

        Map<String, Object> b = new HashMap<>();
        b.put("nombre", "Andres R");
        b.put("Edad", 26);

        Map<String, Object> c = new HashMap<>();
        c.put("nombre", "Anyelo");
        c.put("Edad", 30);

        Map<String, Object> d = new HashMap<>();
        d.put("nombre", "Esteban A");
        d.put("Edad", 28);

        Map<String, Object> e = new HashMap<>();
        e.put("nombre", "Felipe");
        e.put("Edad", 23);

        Map<String, Object> f = new HashMap<>();
        f.put("nombre", "Fito");
        f.put("Edad", 32);

        Map<String, Object> g = new HashMap<>();
        g.put("nombre", "Hanssy R");
        g.put("Edad", 27);

        Map<String, Object> h = new HashMap<>();
        h.put("nombre", "Jhon");
        h.put("Edad", 26);

        Map<String, Object> i = new HashMap<>();
        i.put("nombre", "juan pablo");
        i.put("Edad", 25);

        Map<String, Object> j = new HashMap<>();
        j.put("nombre", "Nilo");
        j.put("Edad", 26);

        Map<String, Object> k = new HashMap<>();
        k.put("nombre", "Santiago");
        k.put("Edad", 20);

        Map<String, Object> l = new HashMap<>();
        l.put("nombre", "Santiago s");
        l.put("Edad", 28);

        Map<String, Object> m = new HashMap<>();
        m.put("nombre", "Yuri");
        m.put("Edad", 28);

       
        personas.add(a);
        personas.add(b);
        personas.add(c);
        personas.add(d);
        personas.add(e);
        personas.add(f);
        personas.add(g);
        personas.add(h);
        personas.add(i);
        personas.add(j);
        personas.add(k);
        personas.add(l);
        personas.add(m);

       
        for (Map<String, Object> persona : personas) {
            System.out.println(persona.get("nombre") + " " + persona.get("Edad"));
        }

        
        for (Map<String, Object> persona : personas) {
            int edad = (int) persona.get("Edad");
            
            persona.put("Edad", edad + 1);
        }

       
        System.out.println("\nDespués de operar con los valores de edad:");
        for (Map<String, Object> persona : personas) {
            System.out.println(persona.get("nombre") + " " + persona.get("Edad"));
        }
    }
}

File: ruby/bucles-for.rb
# For loop en Ruby para imprimir los nombres
nombres = ["camila", "sebastian R", "Cristian", "Andres R", "Anyelo", "Esteban A", "Felipe", "Fito", "Hanssy R", "Hanz l", "Jhon", "juan pablo", "Nilo", "Santiago", "Santiago S", "Yuri \n\n"]

# bucle for en Ruby
for nombre in nombres
  puts nombre
end

# Diccionarios en Ruby (Hashes)
personas = [] # Lista vacía

# Definir los diccionarios (Hashes)
a = {'nombre' => 'sebastian R', 'Edad' => 24}
b = {'nombre' => 'Andres R', 'Edad' => 26}
c = {'nombre' => 'Anyelo', 'Edad' => 30}
d = {'nombre' => 'Esteban A', 'Edad' => 28}
e = {'nombre' => 'Felipe', 'Edad' => 23}
f = {'nombre' => 'Fito', 'Edad' => 32}
g = {'nombre' => 'Hanssy R', 'Edad' => 27}
h = {'nombre' => 'Jhon', 'Edad' => 26}
i = {'nombre' => 'juan pablo', 'Edad' => 25}
j = {'nombre' => 'Nilo', 'Edad' => 26}
k = {'nombre' => 'Santiago', 'Edad' => 20}
l = {'nombre' => 'Santiago s', 'Edad' => 28}
m = {'nombre' => 'Yuri', 'Edad' => 28}

# Añadir los hashes a la lista
personas << a
personas << b
personas << c
personas << d
personas << e
personas << f
personas << g
personas << h
personas << i
personas << j
personas << k
personas << l
personas << m

# Imprimir nombres y edades
for persona in personas
  puts "#{persona['nombre']}, #{persona['Edad']}"
end

# Operar con los valores de los hashes, aumentar la edad de todas las personas en 1
personas.each do |persona|
  persona['Edad'] += 1
end

# Imprimir la nueva edad de cada persona
puts "\nEdades después de aumentar en 1 año:"
for persona in personas
  puts "#{persona['nombre']}, #{persona['Edad']}"
end
